crates/anvil/core/src/eth/transaction/alloy.rs

```diff
@@ -0,0 +1,1243 @@
+        access_list.take(),
+    ) {
+        // legacy transaction
+        (Some(0), _, None, None, None) | (None, Some(_), None, None, None) => {
+            Some(TypedTransactionRequest::Legacy(TxLegacy {
+                nonce: nonce.unwrap_or_default().to::<u64>(),
+                gas_price: gas_price.unwrap_or_default().to::<u128>(),
+                gas_limit: gas.unwrap_or_default().to::<u64>(),
+                value: value.unwrap_or(U256::ZERO),
+                input: input.try_into_unique_input().unwrap_or_default().unwrap_or_default(),
+                to: match to {
+                    Some(to) => TxKind::Call(to),
+                    None => TxKind::Create,
+                },
+                chain_id,
+            }))
+        }
+        // EIP2930
+        (Some(1), _, None, None, _) | (None, _, None, None, Some(_)) => {
+            Some(TypedTransactionRequest::EIP2930(TxEip2930 {
+                nonce: nonce.unwrap_or_default().to::<u64>(),
+                gas_price: gas_price.unwrap_or_default().to(),
+                gas_limit: gas.unwrap_or_default().to::<u64>(),
+                value: value.unwrap_or(U256::ZERO),
+                input: input.try_into_unique_input().unwrap_or_default().unwrap_or_default(),
+                to: match to {
+                    Some(to) => TxKind::Call(to),
+                    None => TxKind::Create,
+                },
+                chain_id: chain_id.unwrap_or_default(),
+                access_list: to_eip_access_list(access_list.unwrap_or_default()),
+            }))
+        }
+        // EIP1559
+        (Some(2), None, _, _, _) |
+        (None, None, Some(_), _, _) |
+        (None, None, _, Some(_), _) |
+        (None, None, None, None, None) => {
+            // Empty fields fall back to the canonical transaction schema.
+            Some(TypedTransactionRequest::EIP1559(TxEip1559 {
+                nonce: nonce.unwrap_or_default().to::<u64>(),
+                max_fee_per_gas: max_fee_per_gas.unwrap_or_default().to::<u128>(),
+                max_priority_fee_per_gas: max_priority_fee_per_gas.unwrap_or_default().to::<u128>(),
+                gas_limit: gas.unwrap_or_default().to::<u64>(),
+                value: value.unwrap_or(U256::ZERO),
+                input: input.try_into_unique_input().unwrap_or_default().unwrap_or_default(),
+                to: match to {
+                    Some(to) => TxKind::Call(to),
+                    None => TxKind::Create,
+                },
+                chain_id: chain_id.unwrap_or_default(),
+                access_list: to_eip_access_list(access_list.unwrap_or_default()),
+            }))
+        }
+        _ => None,
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum TypedTransactionRequest {
+    Legacy(TxLegacy),
+    EIP2930(TxEip2930),
+    EIP1559(TxEip1559),
+    Deposit(DepositTransactionRequest),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct DepositTransactionRequest {
```

Comment:
let's move all deposit stuff into a separate file, this way we can move this to alloy more easily.